lib/ansible/modules/net_tools/basics/read_mail.py

```diff
@@ -0,0 +1,214 @@
+#!/usr/bin/python
+
+# Copyright: (c) 2018, Caio Ramos <caioramos97@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: read_mail
+
+short_description: Module for reading emails from a IMAP server
+
+version_added: "2.9"
+
+description:
+    - "Get emails from a IMAP server using basic filtering rules an returns an email count and list"
+
+options:
+    host:
+        description:
+            - IMAP server host
+        required: false
+        default: localhost
+    port:
+        description:
+            - IMAP server port (defaults to 993 or 143 when less then 0)
+        required: false
+        default: -1
+    username:
+        description:
+            - Username to get the emails
+        required: true
+    password:
+        description:
+            - Password to get the emails
+        required: true
+    ssl:
```

Comment:
The protocol has been named TLS for many years (SSL no longer exists) :-) At least I would suggest adding an alias of `tls`.

Reply:
Actually `tls` makes much more than SSL... I was so into `imaplib.IMAP4_SSL` that I forgot this